src/syft/core/common/event_loop.py

```diff
@@ -0,0 +1,115 @@
+# stdlib
+import asyncio
+import atexit
+import os
+from typing import Any as TypeAny
+from typing import Dict as TypeDict
+from typing import Optional
+
+# third party
+import nest_asyncio
+
+# syft relative
+from .environment import is_interactive
+from .environment import is_jupyter
+
+# Depending on where syft is imported we may need to get the existing event loop or
+# create a new one. In the case of Jupyter there is an existing event loop created
+# which we need to get otherwise our events might get blocked. If we import via CLI
+# there will be no event loop so we need to create one here and set it or the network
+# code will try to create its own and in some instances they wont match
+
+try:
+    # stdlib
+    from asyncio import get_running_loop  # noqa Python >=3.7
+except ImportError:  # pragma: no cover
+    # stdlib
+    from asyncio.events import _get_running_loop as get_running_loop  # pragma: no cover
+
+# first lets get any existing event loop, this will throw an exception in if there is
+# no event loop or will return none in older versions of python
+loop = None
+try:
+    loop = get_running_loop()
+except RuntimeError:
+    pass
+
+
+# no event loop found, lets create one so that future asyncio imports use the same loop
+if loop is None:
+    loop = asyncio.new_event_loop()
+
+# set the event loop so its the same everywhere
+asyncio._set_running_loop(loop)
+
+
+# https://github.com/erdewit/nest_asyncio
+# patch the event loop to allow nested event loops
+try:
+    nest_asyncio.apply(loop)
+except Exception as e:
+    print(f"Failed to patch event loop with nest_asyncio {e}")
```

Comment:
Q: Shouldn't we raise the exception?

Reply:
I added this to try to catch an exception which happens sometimes in python 3.6.
```
Exception in callback Task._wakeup(<Future finis...2.253.59.127'>)
handle: <Handle Task._wakeup(<Future finis...2.253.59.127'>)>
Traceback (most recent call last):
  File "/Users/madhavajay/.pyenv/versions/3.6.9/lib/python3.6/asyncio/events.py", line 145, in _run
    self._callback(*self._args)
  File "/Users/madhavajay/.pyenv/versions/3.6.9/lib/python3.6/asyncio/tasks.py", line 276, in _wakeup
    self._step()
  File "/Users/madhavajay/.local/share/virtualenvs/PySyft-lHlz_cKe/lib/python3.6/site-packages/nest_asyncio.py", line 159, in step
    step_orig(task, exc)
  File "/Users/madhavajay/.pyenv/versions/3.6.9/lib/python3.6/asyncio/tasks.py", line 260, in _step
    self.__class__._current_tasks.pop(self._loop)
KeyError: <_UnixSelectorEventLoop running=True closed=False debug=False>
```

It seems to be related to nest_asyncio patching asyncio and possibly due to us using the event loop in a new thread. The try except doesn't catch it nor does one inside the thread code. I tried a bunch of ways to catch or prevent this exception but I cant. It also seems to have no negative effect on the application running so since its for python 3.6 im just going to leave it for now.

I have removed the try except because theres nothing for it to catch anyway.